hoa-backend/src/auth/dto/custom-exception.filter.ts

```diff
@@ -0,0 +1,25 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Response } from 'express';
+
+@Catch(NotFoundException, UnauthorizedException)
+export class CustomExceptionFilter implements ExceptionFilter {
```

Comment:
hmm interesting, what does this do?

Reply:
this is your customized way of handling exceptions. so instead of like throw new HttpException(
        'Email address is already in use',
        HttpStatus.CONFLICT,
      );